File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/composite/Node.java
package patterns.composite;

public abstract class Node { 
    public Node(String name) {
        super();
        this.name = name;
    }
    public String toString() {
        return name;
    }
    public void print() {
        print(0);
    }
    public abstract long length();
    protected abstract void print(int depth);
    protected void printPrefix(int depth) {
        for(int i=0;i<depth;i++) {
            System.out.print("----");
        }
        System.out.print("->");
    }
    protected String name;
}

File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/Composite/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Composite {
    internal abstract class Node {
        public Node(string name) {
            this.name = name;
        }

        public override string ToString() {
            return name;
        }

        public void Print() {
            Print(0);
        }

        public abstract long Length { get; }
        public abstract void Print(int depth);

        protected void PrintPrefix(int depth) {
            for (int i = 0; i < depth; i++) {
                Console.Write("----");
            }
            Console.Write("->");
        }

        protected String name;
    }
}